packages/story-editor/src/components/mediaRecording/footer.js

```diff
@@ -0,0 +1,253 @@
+import { Z_INDEX_RECORDING_MODE } from '../../constants/zIndex';
+import { FooterArea } from '../canvas/layout';
+import useUploadWithPreview from '../canvas/useUploadWithPreview';
+import useMediaRecording from './useMediaRecording';
+import { COUNTDOWN_TIME_IN_SECONDS } from './constants';
+
+const StyledFooter = styled(FooterArea)`
+  display: flex;
+  align-items: start;
+  flex-direction: row;
+  justify-content: center;
+  z-index: ${Z_INDEX_RECORDING_MODE};
+`;
+
+const BaseButton = styled(Button).attrs({
+  type: BUTTON_TYPES.PRIMARY,
+  size: BUTTON_SIZES.SMALL,
+})``;
+
+const RecordingButton = styled(BaseButton)`
+  margin-right: 20px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+
+  svg {
+    margin: -9px 10px -9px 0;
+  }
+`;
+
+const StopButton = styled(BaseButton)`
+  margin-right: 20px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+
+  background-color: ${({ theme }) => theme.colors.interactiveBg.negativeNormal};
+  border-color: ${({ theme }) => theme.colors.interactiveBg.negativeNormal};
+  color: ${({ theme }) => theme.colors.gray[5]};
+
+  &:active,
+  &:hover,
+  &:focus {
+    background-color: ${({ theme }) =>
+      theme.colors.interactiveBg.negativeHover};
+    border-color: ${({ theme }) => theme.colors.interactiveBg.negativeHover};
+  }
+`;
+
+const CaptureButton = styled(BaseButton).attrs({
+  type: BUTTON_TYPES.QUATERNARY,
+})`
+  background-color: ${({ theme }) => theme.colors.opacity.overlayDark};
+  padding-top: 12px;
+  padding-bottom: 12px;
+
+  svg {
+    margin: -9px 10px -9px 0;
+  }
+`;
+
+const InsertButton = styled(Button).attrs({
+  type: BUTTON_TYPES.PRIMARY,
+  size: BUTTON_SIZES.SMALL,
+})``;
+
+const RetryButton = styled(Button).attrs({
+  type: BUTTON_TYPES.PLAIN,
+  size: BUTTON_SIZES.SMALL,
+})`
+  margin-right: 20px;
+`;
+
+function Footer({ captureImage }) {
+  const {
+    status,
+    file,
+    isGif,
+    isMuted,
+    hasMediaToInsert,
+    countdown,
+    duration,
+    isCountingDown,
+    setMediaBlobUrl,
+    setDuration,
+    setCountdown,
+    startRecording,
+    stopRecording,
+    setFile,
+    toggleRecordingMode,
+    getMediaStream,
+    resetState,
+  } = useMediaRecording(({ state, actions }) => ({
+    status: state.status,
+    file: state.file,
+    isGif: state.isGif,
+    isMuted: !state.hasAudio || state.isGif,
+    countdown: state.countdown,
+    duration: state.duration,
+    isCountingDown: state.isCountingDown,
+    hasMediaToInsert: Boolean(state.mediaBlobUrl),
+    startRecording: actions.startRecording,
+    stopRecording: actions.stopRecording,
+    setFile: actions.setFile,
+    setMediaBlobUrl: actions.setMediaBlobUrl,
+    setDuration: actions.setDuration,
+    setCountdown: actions.setCountdown,
+    toggleRecordingMode: actions.toggleRecordingMode,
+    getMediaStream: actions.getMediaStream,
+    resetState: actions.resetState,
+  }));
+
+  const uploadWithPreview = useUploadWithPreview();
+
+  const onRetry = useCallback(() => {
+    resetState();
```

Comment:
Noting:
When using "image" mode.

I was expecting when I pressed retry I would remain in "image" mode ... with a fresh countdown vs being taken back to the record video / capture image option screen.

Reply:
What if you decided that you don't want to take a photo anymore but instead record a video?

Reply:
True - I can exit but that's quite a few more clicks to get back to either type of record mode vs going back to the options buttons.  Thinking more of a user doing this multiple times trying to get the image they want.  

I think what you have works works for now.